File: src/main/java/phoenix/world/builders/Builders.java
package phoenix.world.builders;

import net.minecraft.block.Blocks;
import net.minecraft.world.gen.surfacebuilders.DefaultSurfaceBuilder;
import net.minecraft.world.gen.surfacebuilders.SurfaceBuilder;
import net.minecraft.world.gen.surfacebuilders.SurfaceBuilderConfig;
import phoenix.init.PhoenixBlocks;

public class Builders
{
    //Билдеры
    public static final SurfaceBuilder<AdvansedSurfaceBuilderConfig> UNDER = SurfaceBuilder.register("under", new UnderSurfaceBuilder(AdvansedSurfaceBuilderConfig::deserialize));
    public static final SurfaceBuilder<SurfaceBuilderConfig> HEARTVOID = SurfaceBuilder.register("heart", new HeartVoidSurfaceBuilder(SurfaceBuilderConfig::deserialize));

    //Конфиги для билдеров
    public static final AdvansedSurfaceBuilderConfig UNDER_CONFIG     = new AdvansedSurfaceBuilderConfig(Blocks.OBSIDIAN, Blocks.OBSIDIAN, Blocks.OBSIDIAN, PhoenixBlocks.FERTILE_END_STONE.get());
    public static final SurfaceBuilderConfig         HEARTVOID_CONFIG = new SurfaceBuilderConfig(Blocks.END_STONE.getDefaultState(), Blocks.END_STONE.getDefaultState(), Blocks.END_STONE.getDefaultState());

}

File: src/main/java/phoenix/world/builders/Builders.kt
package phoenix.world.builders

import com.mojang.datafixers.Dynamic
import net.minecraft.block.Blocks
import net.minecraft.world.gen.surfacebuilders.SurfaceBuilder
import net.minecraft.world.gen.surfacebuilders.SurfaceBuilderConfig
import phoenix.init.PhoenixBlocks

object Builders
{
    //Билдеры
    val UNDER: SurfaceBuilder<AdvancedSurfaceBuilderConfig> = SurfaceBuilder.register("under", UnderSurfaceBuilder { p_215455_0_: Dynamic<*>? -> AdvancedSurfaceBuilderConfig.deserialize(p_215455_0_) })
    val HEARTVOID: SurfaceBuilder<SurfaceBuilderConfig> = SurfaceBuilder.register("heart", HeartVoidSurfaceBuilder { p_215455_0_: Dynamic<*>? -> SurfaceBuilderConfig.deserialize(p_215455_0_) })

    //Конфиги для билдеров
    val UNDER_CONFIG = AdvancedSurfaceBuilderConfig(
        Blocks.END_STONE,
        Blocks.END_STONE,
        Blocks.END_STONE,
        PhoenixBlocks.FERTILE_END_STONE.get()
    )
    val HEARTVOID_CONFIG = SurfaceBuilderConfig(Blocks.END_STONE.defaultState, Blocks.END_STONE.defaultState, Blocks.END_STONE.defaultState)
}